docs/core-concepts/bot-sequences/actions.mdx

```diff
@@ -1,9 +1,624 @@
+                ]
+            }
+        }
+    }
+}
+```
+
+### Behaviour
+
+The `Behaviour` action type runs a MonoBehaviour. This can be useful for running agent code that relies on
+particular data and information from the code of your game, such as tapping into an existing navigation system.
+
+**Data Format**
+
+`type` is set to `Behaviour`.
+
+```json
+// The `botAction.data` format
+{
+    "behaviourFullName": "...", // The full class path of the MonoBehaviour to run
+    "maxRuntimeSeconds": ... // The maximum amount of time to run the behaviour for in seconds before it is stopped
+}
+```
+
+The Behaviour itself needs to be defined in the codebase of your game where it can be found during runtime. It should be
+a class that inherits from `MonoBehaviour`. In order to terminate the behaviour, you can use `Destroy(this)` to destroy the
+behaviour instance.
+
+**Example: Running a pathfinding behaviour for at max 30 seconds**
+
+```csharp
+// The monobehaviour code contained somewhere in your game ruintime code
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace RG.Example {
+
+    public class NavigateToPosition : MonoBehaviour
+    {
+        private NavMeshAgent agent;
+        public Vector3 targetPosition;
+
+        private void Start()
+        {
+            agent = GetComponent<NavMeshAgent>();
+            if (agent != null)
+            {
+                agent.SetDestination(targetPosition);
+            }
+        }
+
+        private void Update()
+        {
+            if (agent != null && !agent.pathPending)
+            {
+                if (agent.remainingDistance <= agent.stoppingDistance)
+                {
+                    if (!agent.hasPath || agent.velocity.sqrMagnitude == 0f)
+                    {
+                        Destroy(this);
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+```json
+{
+    "name": "Run pathfinding behaviour",
+    "description": "Runs a pathfinding behaviour to navigate to a position",
+    "endCriteria": [...],
+    "botAction": {
+        "type": "Behaviour",
+        "data": {
+            "behaviourFullName": "RG.Example.NavigateToPosition",
+            "maxRuntimeSeconds": 30.0
+        }
+    }
+}
+```
+
+### MonkeyBot
+
+The `MonkeyBot` action type runs a chaos monkey bot for a given amount of time. This can be useful for testing the robustness of your game.
+This Monkey Bot will randomly select actions to perform from a list of possible actions, and will perform them at a given interval. These
+action are found via a code analysis of the game, which must be run before you use this segment type.
+
+**Setup**
+
+In order to setup the monkey bot, you need to run a code analysis of the game. This can be done from the **Regression Games > Configure Bot Actions** menu.
+
+Once you click that menu option, a panel will appear. Click the "Analyze Actions" button, which will perform an analysis of your game and
```

Comment:
We should add a sentence about them filtering out actions.

Reply:
Added a note about this. The experience here is actually pretty bad.... since Monkey Bots are configured in the botAction JSON now, these settings in the panel are completely ignored (and this is a low hanging fruit I wanted to address soon).